parser.ts

```diff
@@ -185,6 +185,21 @@ export function traverseExpr(c : TreeCursor, s : string) : Expr<null> {
         tag: "id",
         name: "self"
       };
+    case "ArrayExpression":
```

Comment:
make sure you always have some instruction within the cases even if you don't know the implementation at the moment, for example raise an error or warning message for now, until later fill the gaps. it would be nice to leave tiny comments of a tiny example of what you are parsing in each case beneath it.

Reply:
In fact, I intentionally left this case empty in order to fall-through to the next case, TupleExpression, which has the same logic. I am sorry that it made you confused. I have added some comments.